### PR TITLE
wireguard-tools, wireguard-go: add update scripts, wireguard-go: 0.0.20181222 -> 0.0.20190517

### PR DESCRIPTION
**Pull request by flokli**
###### Motivation for this change
This adds updateScripts for both wireguard-tools and wireguard-go. As the kernel uses the src from wireguard-tools, this also applies to wireguard kernel modules.

I verified both scripts by updating to the latest version. `wireguard-tools` was already up to date, so I manually reverted the bump. `wireguard-go` was outdated, so I added a commit that updates the package.

I didn't build wireguard-go, will let @GrahamcOfBorg handle that.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

cc @zx2c4

### CODE REVIEW

File: pkgs/tools/networking/wireguard-go/deps.nix [hunk 1]
```diff
@@ -1,30 +1,57 @@
 # file generated from go.mod using vgo2nix (https://github.com/adisbladis/vgo2nix)
 [
+  {
+    goPackagePath = "github.com/Microsoft/go-winio";
+    fetch = {
+      type = "git";
+      url = "https://github.com/Microsoft/go-winio";
```
Comment (zx2c4) [inline pkgs/tools/networking/wireguard-go/deps.nix]: Looks like vgo2nix doesn't respect go.mod's replace operator. Shouldn't be an issue in this case, but it is notable.

File: pkgs/tools/networking/wireguard-go/update.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+#!nix-shell -i bash -p curl gnused common-updater-scripts vgo2nix
+
+set -eu -o pipefail
+
+readonly BASEDIR="$(dirname $(readlink -f $0))"
+
+version="$(curl -sL https://build.wireguard.com/distros.txt | sed -n 's/^upstream\tgo\t\([^\t]\+\)\t.*/\1/p')"
+update-source-version wireguard-go "$version"
+
+vgo2nix -dir $(nix-build -A wireguard-go.src) -outfile $PWD/pkgs/tools/networking/wireguard-go/deps.nix
```
Comment (flokli) [inline pkgs/tools/networking/wireguard-go/update.sh]: I'm not sure if it's right to assume `$PWD` to be like this, or if there's a better way to do it -  I couldn't find a good example for that. /cc @jtojnar

File: pkgs/tools/networking/wireguard-go/update.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+if [[ -f $PWD/wireguard-go.log ]];then
+    rm $PWD/wireguard-go.log
```
Comment (flokli) [inline pkgs/tools/networking/wireguard-go/update.sh]: @adisbladis can we teach vgo2nix to not write a log file?

**Comment by flokli** [comment]
@GrahamcOfBorg build wireguard-go

**Review by zx2c4 — APPROVED** [review]
Thanks!

File: pkgs/tools/networking/wireguard-go/update.sh (hunk 3 above)
Comment (flokli) [reply pkgs/tools/networking/wireguard-go/update.sh]: Opened: https://github.com/adisbladis/vgo2nix/issues/21

File: pkgs/tools/networking/wireguard-go/deps.nix (hunk 1 above)
Comment (flokli) [reply pkgs/tools/networking/wireguard-go/deps.nix]: Could you open an issue at https://github.com/adisbladis/vgo2nix for that?

I lack the go skills to investigate this further.

File: pkgs/tools/networking/wireguard-go/update.sh (hunk 2 above)
Comment (jtojnar) [reply pkgs/tools/networking/wireguard-go/update.sh]: Currently, this is undefined, see https://github.com/NixOS/nixpkgs/issues/61935, using `git rev-parse --show-toplevel` is probably more safe bet.

Comment (flokli) [reply pkgs/tools/networking/wireguard-go/update.sh]: Updated it to use that, thanks!

**Review by fpletz — APPROVED** [review]

File: pkgs/tools/networking/wireguard-go/deps.nix (hunk 1 above)
Comment (flokli) [reply pkgs/tools/networking/wireguard-go/deps.nix]: Opened issue upstream: https://github.com/adisbladis/vgo2nix/issues/22

**Pull request merged**